File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        string push = "Push",pop = "Pop";
        vector<string>res;
        for(int i = 1,j = 0;i<=n && j < target.size();i++){
            if(i == target[j]){
                res.push_back(push);
                j++;
            }
            else{
                res.push_back(push);
                res.push_back(pop);
            }
        }
        return res;
    }
};

File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        String push = "Push",pop = "Pop";
        List<String>res = new ArrayList<String>();
        for(int i = 1,j = 0;i<=n && j < target.length;i++){
            if(i == target[j]){
                res.add(push);
                j++;
            }
            else{
                res.add(push);
                res.add(pop);
            }
        }
        return res;
    }
}